core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -858,19 +844,33 @@ object SparkSubmit extends CommandLineUtils {
     require(path != null, "path cannot be null.")
     val uri = Utils.resolveURI(path)
     uri.getScheme match {
-      case "file" | "local" =>
-        path
-
+      case "file" | "local" => path
+      case "https" | "http" | "ftp" => path
```

Comment:
This is wrong, right? These need to be downloaded, and now they won't be.

(Whether the `FileSystem` API can download those is a separate question... if it doesn't, you'll end up with another implementation of `Utils.doFetchFile`.)